src/model.rs

```diff
@@ -999,22 +999,25 @@ impl Model {
     /// # use transit_model::objects::Transfer;
     /// let mut collections = Collections::default();
     /// // This transfer is invalid as there is no stop points in collections
+    /// // but objects not referenced are removed from the model
     /// collections.transfers = Collection::from(Transfer {
     ///     from_stop_id: "invalid".into(),
     ///     to_stop_id: "also_invalid".into(),
     ///     min_transfer_time: None,
     ///     real_min_transfer_time: None,
     ///     equipment_id: None,
     /// });
-    /// assert!(Model::new(collections).is_err());
+    /// assert!(Model::new(collections).is_ok());
     /// ```
     pub fn new(mut c: Collections) -> Result<Self> {
-        fn apply_generic_business_rules(collections: &mut Collections) {
+        fn apply_generic_business_rules(collections: &mut Collections) -> Result<()> {
+            collections.sanitize()?;
```

Comment:
The ‘sanitize’ must be done in last. For example, ‘enhance_route_name’ might skip some lines if not able to generate a ‘route_name’. Therefore, some ‘trip’ might need to be sanitized.

Reply:
ok. i might write in this order
route_name
sanitize
co2 and the other

to avoid useless processes objects that will removed

Reply:
I wouldn't care so much for the order performance until it is really a problem. Let's prove it is a problem before doing such optimizations. For now, I would keep it simple and put it at the end.

Reply:
done